src/productshandler.cpp

```diff
@@ -44,6 +44,11 @@ ProductsHandler::ProductsHandler(QObject* parent) : QAbstractListModel(parent) {
   Q_ASSERT(!s_instance);
   s_instance = this;
 
+  // If web purchases are supported, then we don't need to do anything.
```

Comment:
Why move this in?

Reply:
I couldn't find a way to conditionally register a singleton type purely using `qt_add_qml_module()` in a way that would respect the feature flags so I figured the cleanest solution was to register all the QML types in the `Mozilla.VPN` module (and to do so consistently regardless of feature flagging).

In order to replicate the intended behaviour of the C++ code that this replaces, it was necessary to ensure that the `ProductsHandler` class doesn't do anything when the feature flag is disabled even if we happen to instantiate it.